File: src/main/java/com/jaasielsilva/helpdesk/service/UsuarioService.java
package com.jaasielsilva.helpdesk.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.security.core.Authentication;

import com.jaasielsilva.helpdesk.dto.usuario.CreateUsuarioRequest;
import com.jaasielsilva.helpdesk.dto.usuario.UpdateUsuarioRequest;
import com.jaasielsilva.helpdesk.dto.usuario.UsuarioResponse;

public interface UsuarioService {

    Page<UsuarioResponse> listar(Long empresaId, Pageable pageable, Authentication auth);

    UsuarioResponse buscarPorId(Long id, Authentication auth);

    UsuarioResponse criar(CreateUsuarioRequest request, Authentication auth);

    UsuarioResponse atualizar(Long id, UpdateUsuarioRequest request, Authentication auth);

    void desativar(Long id, Authentication auth);
}

File: frontend/src/app/core/services/usuario.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable, map } from 'rxjs';

import {
  CreateUsuarioRequest,
  UpdateUsuarioRequest,
  Usuario
} from '../models/usuario';
import { PageResponse } from '../models/page-response';

@Injectable({ providedIn: 'root' })
export class UsuarioService {
  private readonly http = inject(HttpClient);

  listar(page = 0, size = 10, empresaId?: number | null): Observable<PageResponse<Usuario>> {
    let params = new HttpParams()
      .set('page', page)
      .set('size', size)
      .set('sort', 'dataCriacao,desc');

    if (empresaId != null) {
      params = params.set('empresaId', empresaId);
    }

    return this.http.get<{ dados: PageResponse<Usuario> }>('/api/usuarios', { params }).pipe(
      map((res) => res.dados)
    );
  }

  criar(request: CreateUsuarioRequest): Observable<Usuario> {
    return this.http.post<{ dados: Usuario }>('/api/usuarios', request).pipe(
      map((res) => res.dados)
    );
  }

  atualizar(id: number, request: UpdateUsuarioRequest): Observable<Usuario> {
    return this.http.put<{ dados: Usuario }>(`/api/usuarios/${id}`, request).pipe(
      map((res) => res.dados)
    );
  }

  desativar(id: number): Observable<void> {
    return this.http.delete<{ dados: void }>(`/api/usuarios/${id}`).pipe(
      map(() => undefined)
    );
  }
}
